service/src/main/java/skills/storage/repos/QuizToSkillDefRepo.groovy

```diff
@@ -50,6 +54,21 @@ interface QuizToSkillDefRepo extends JpaRepository<QuizToSkillDef, Long> {
             group by q.quizId''')
     QuizNameAndId getQuizIdBySkillIdRef(Integer skillIdRef)
 
+    @Nullable
+    @Query('''select child.name as skillName, child.skillId as skillId, child.projectId as projectId, subject.skillId as subjectId,
+                     subject.name as subjectName, exists(
+                       select ur.roleName from UserRole ur where ur.projectId = child.projectId and
+                       ur.roleName in ('ROLE_PROJECT_ADMIN', 'ROLE_QUIZ_ADMIN') and ur.userId = ?2
```

Comment:
I think this needs to be ``ur.roleName in ('ROLE_PROJECT_ADMIN', 'ROLE_PROJECT_APPROVER')``; in addition an `OR`  if user has ``ROLE_SUPER_DUPER_USER`` role; will need service tests to validate as well